base/broadcast.jl

```diff
@@ -25,13 +25,14 @@ function broadcast!{T,S,N}(::typeof(identity), x::AbstractArray{T,N}, y::Abstrac
 end
 
 # logic for deciding the resulting container type
-containertype(x) = containertype(typeof(x))
+containertype(x) = _containertype(typeof(x))
 containertype(::Type) = Any
```

Comment:
For clarity, this should perhaps be reordered so that all `containertype` methods are together, after all the `_containertype` methods.